### ISSUE TITLE
build.sbt publishes required nscplugin too many times

### ISSUE THREAD

**Issue by LeeTibbert** [scala-native/scala-native#1803]
I have been watching a lot of build log files the past day or so as I rebase many of my PRs.
I strongly believe, but have not yet proven to myself, that PR #1782 introduced a bug
where the required nscplugin is getting published too many times.  To many is better than
too few, but, unfortunately, it means that the build time increases.

I believe the fix is to change the current unconditional  `nscplugin/publishLocal` to condition
on the absence of the nscplugin artifact.

My apology to anyone spending extra time on builds.

**Comment by LeeTibbert** [comment]
Agreed, trying to fold the Scala.js dependency approach into the SN build is almost
certainly the mid-range solution or on the path to it.

In the ultra-short range can I assure myself and the community that PR #1782 is no worse
that the day before it was accepted?

While waiting for full build to complete after I rebase my PRs, I searched 
log files from today and from 9 months previous for a ""rebuild ; test-all"
cycle.

nscplugin was published 4 (!) times in each case; two in rebuild & two in test-all.
I would expect that plugin to be published once.  The second publication in rebuild is
not good, but also not all that bad.  I found the two publications in test-all to be
more than a bit astonishing, but not the end of the world.  My eye must have been picking
up the excess publications in test-all and blaming PR #1782 for them.

I can try the same experiment with log files from just before PR #1782 was merged.
I expect the same results, but surprises are why one runs experiments.

It is not a goal of this Issue, but something else must be increasing my build time.

More information as I continue tracing.
